src/library_fs.js

```diff
@@ -1126,6 +1126,9 @@ FS.staticInit();
       if (stream.stream_ops.open) {
         stream.stream_ops.open(stream);
       }
+      if (created) {
+        FS.chmod(node, mode & 0o777);
+      }
```

Comment:
Why not just rely on `mknod` to set the correct mode above?  It seems odd to explicitly call `mknod` with 777 and then call `chmod` afterwards, no?

Reply:
Because then if we call `open("some_file", O_CREAT, 0x000)` in the nodefs `stream.stream_ops.open(stream)` fails. We call `mknod` to produce a native node with no permissions and then separately try to open it but we didn't give ourselves permission to open it. Instead, for this to work on nodefs we either need to detect this situation and make a single native call to `open("some_file", O_CREAT, 0x000)` and use the file descriptor it returns or to make the change I did here.